### ISSUE TITLE
Try this architecture in the speech enhancement for speech recognition?

### ISSUE THREAD

**Issue by sigpro** [andabi/music-source-separation#2]
Can this architecture recover speech from the noisy speech?

**Comment by andabi** [comment]
@sigpro 
The same architecture can be used for removing noise from speech or extracting single speech from dialog as well ;)
Take a look at this reference paper: https://posenhuang.github.io/papers/Joint_Optimization_of_Masks_and_Deep%20Recurrent_Neural_Networks_for_Monaural_Source_Separation_TASLP2015.pdf

**Comment by sigpro** [comment]
@andabi 
Thank you for reply,will you try multi-channel for better performance?

**Comment by andabi** [comment]
@sigpro yes i'll it if have some dataset ;) Do you know any dataset that has multi-channel ?

**Comment by sigpro** [comment]
@andabi I think CHiME challenge's dataset can be used.

**Comment by dankorg** [comment]
Just to extend this question. How do I modify the code to make this work with any dataset? Or is the code specifically created for singing voice separation? Would be awesome if this is a universal solution to many source separation tasks.
